File: src/main/java/com/dogGetDrunk/meetjyou/common/exception/ErrorResponse.java
package com.dogGetDrunk.meetjyou.common.exception;

import lombok.Getter;

import java.util.ArrayList;
import java.util.List;

@Getter
public class ErrorResponse {

    private int status;
    private String message;
    private List<String> values = new ArrayList<>();

    public ErrorResponse(int status, ErrorCode errorCode) {
        this.status = status;
        this.message = errorCode.getMessage();
    }

    public ErrorResponse(int status, ErrorCode errorCode, String value) {
        this(status, errorCode);
        this.values = List.of(value);
    }

    public ErrorResponse(int status, ErrorCode errorCode, List<String> values) {
        this(status, errorCode);
        this.values = values;
    }
}

File: src/main/java/com/dogGetDrunk/meetjyou/common/exception/ErrorResponse.kt
package com.dogGetDrunk.meetjyou.common.exception

data class ErrorResponse(
    val status: Int,
    val message: String,
    val values: List<String> = emptyList()
) {
    constructor(status: Int, errorCode: ErrorCode) : this(
        status,
        errorCode.message
    )

    constructor(status: Int, errorCode: ErrorCode, value: String) : this(
        status,
        errorCode.message,
        listOf(value)
    )

    constructor(status: Int, errorCode: ErrorCode, values: List<String>) : this(
        status,
        errorCode.message,
        values
    )
}
